src/test/groovy/unit/org/pih/warehouse/api/ProductApiTest.groovy

```diff
@@ -0,0 +1,178 @@
+package unit.org.pih.warehouse.api
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties
+import groovy.transform.builder.Builder
+import io.restassured.RestAssured
+import io.restassured.builder.RequestSpecBuilder
+import io.restassured.http.ContentType
+import io.restassured.http.Cookie
+import io.restassured.path.json.JsonPath
+import io.restassured.specification.RequestSpecification
+import lombok.AllArgsConstructor
+import lombok.Data
+import lombok.NoArgsConstructor
+import org.grails.web.json.JSONObject
+import spock.lang.Shared
+import spock.lang.Specification
+import static io.restassured.RestAssured.*
+import io.restassured.response.Response
+import static org.hamcrest.MatcherAssert.assertThat
+import static org.hamcrest.Matchers.is
+import static org.hamcrest.Matchers.notNullValue
+import static org.hamcrest.Matchers.lessThanOrEqualTo
+
+class ProductApiTest extends Specification {
+
+    @Shared
+    String baseURI = System.getProperty("server.host")?:"http://localhost"
```

Comment:
Please add whitespace after and before the Elvis operators
```suggestion
    String baseURI = System.getProperty("server.host") ?: "http://localhost"
```

Reply:
I actually hate the extra whitespace, but fine.